pandas/core/indexes/datetimelike.py

```diff
@@ -744,6 +752,11 @@ def __sub__(self, other):
         cls.__sub__ = __sub__
 
         def __rsub__(self, other):
+            if is_datetime64_dtype(other) and is_timedelta64_dtype(self):
+                # ndarray[datetime64] cannot be subtracted from self, so
+                # we need to wrap in DatetimeIndex and flip the operation
+                from pandas.core.indexes.datetimes import DatetimeIndex
```

Comment:
since inside a function, can just import from pandas directrly

Reply:
atually pls change all imports that are inside functions to do this (its just shorter)